lnd.go

```diff
@@ -187,16 +187,15 @@ func Main(cfg *Config, lisCfg ListenerCfg, implCfg *ImplementationCfg,
 	// Enable http profiling server if requested.
 	if cfg.Profile != "" {
 		// Create the http handler.
-		profileRedirect := http.RedirectHandler(
-			"/debug/pprof", http.StatusSeeOther,
-		)
-		http.Handle("/", profileRedirect)
+		pprofMux := http.DefaultServeMux
+		http.DefaultServeMux = http.NewServeMux()
```

Comment:
I am not sure if there is any difference between 

```
pprofMux := http.DefaultServeMux
http.DefaultServeMux = http.NewServeMux()

// or
http.DefaultServeMux = http.NewServeMux()
pprofMux := http.DefaultServeMux
```

I guess we are doing it to isolate the pprof endpoints from other traffic without creating a specific mux for it?

Reply:
yea i have very little knowledge in `http` package. Based on the code I think `pprofMux` is a copy of `http.DefaultServerMux`, and we then overwrite `http.DefaultServerMux` with a new mux. Also reversing the order won't work.

Reply:
Yeah, good point. I don't think we should overwrite the default serve mux.
Can't we just do this very explicitly?
Here's the diff I have in mind:
[lnd.diff.txt](https://github.com/lightningnetwork/lnd/files/11815202/lnd.diff.txt)

Reply:
Nice applied the diff!